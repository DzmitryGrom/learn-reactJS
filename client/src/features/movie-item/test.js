import React from 'react';
import { mount } from 'enzyme';
import Item from './index';

const mockDates = {
  "id": 19404,
  "title": "Dilwale Dulhania Le Jayenge",
  "tagline": "Come... Fall in Love",
  "vote_average": 9.2,
  "vote_count": 1297,
  "release_date": "1995-10-20",
  "poster_path": "https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg",
  "overview": "Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancÃ©. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga.",
  "genres": [
    "Comedy",
    "Drama",
    "Romance"
  ],
  "runtime": 190
};

describe('tests items', () => {
  it('render empty', () => {
    const component = mount(<Item />);
    expect(component).toMatchSnapshot();
    expect(component.find('.movieItem__title').text()).toBeFalsy();
    expect(component.find('.movieItem__genre').text()).toBeFalsy();
    expect(component.find('.movieItem__year').text()).toBeFalsy();
    expect(component.find('.movieItem__img').prop('style')).toEqual({"backgroundImage": "url(null)"});
  });
  
  it('render with dates', () => {
    const component = mount(<Item  {...mockDates} />);
    expect(component).toMatchSnapshot();
    expect(component.find('.movieItem__title').text()).toBe('Dilwale Dulhania Le Jayenge');
    expect(component.find('.movieItem__genre').text()).toBe("Comedy Drama Romance ");
    expect(component.find('.movieItem__year').text()).toBe('1995');
    expect(component.find('.movieItem__img').prop('style')).toEqual({"backgroundImage": "url(https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg)"});
  });
})