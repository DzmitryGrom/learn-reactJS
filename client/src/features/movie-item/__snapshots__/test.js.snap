// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests items render empty 1`] = `
<MovieItem
  genres={null}
  poster_path={null}
  release_date={null}
  title={null}
>
  <div
    className="movieItem"
  >
    <div
      className="movieItem__img"
      style={
        Object {
          "backgroundImage": "url(null)",
        }
      }
    />
    <div
      className="movieItem__info"
    >
      <div>
        <h3
          className="movieItem__title"
        />
        <h4
          className="movieItem__genre"
        />
      </div>
      <span
        className="movieItem__year"
      />
    </div>
  </div>
</MovieItem>
`;

exports[`tests items render with dates 1`] = `
<MovieItem
  genres={
    Array [
      "Comedy",
      "Drama",
      "Romance",
    ]
  }
  id={19404}
  overview="Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancÃ©. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga."
  poster_path="https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg"
  release_date="1995-10-20"
  runtime={190}
  tagline="Come... Fall in Love"
  title="Dilwale Dulhania Le Jayenge"
  vote_average={9.2}
  vote_count={1297}
>
  <div
    className="movieItem"
  >
    <div
      className="movieItem__img"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg)",
        }
      }
    />
    <div
      className="movieItem__info"
    >
      <div>
        <h3
          className="movieItem__title"
        >
          Dilwale Dulhania Le Jayenge
        </h3>
        <h4
          className="movieItem__genre"
        >
          <span
            key="Comedy"
          >
            Comedy
             
          </span>
          <span
            key="Drama"
          >
            Drama
             
          </span>
          <span
            key="Romance"
          >
            Romance
             
          </span>
        </h4>
      </div>
      <span
        className="movieItem__year"
      >
        1995
      </span>
    </div>
  </div>
</MovieItem>
`;
